lib/phoenix_component/declarative.ex

```diff
@@ -512,16 +512,19 @@ defmodule Phoenix.Component.Declarative do
     """)
   end
 
-  defp valid_value?(_type, nil), do: true
-  defp valid_value?(:any, _value), do: true
-  defp valid_value?(:string, value), do: is_binary(value)
-  defp valid_value?(:atom, value), do: is_atom(value)
-  defp valid_value?(:boolean, value), do: is_boolean(value)
-  defp valid_value?(:integer, value), do: is_integer(value)
-  defp valid_value?(:float, value), do: is_float(value)
-  defp valid_value?(:list, value), do: is_list(value)
-  defp valid_value?({:struct, mod}, value), do: is_struct(value, mod)
-  defp valid_value?(_type, _value), do: true
+  def valid_value?(_type, nil), do: true
```

Comment:
Can stay private, right?

Reply:
Yes, and most likely can add a test for this specific case without making it public.